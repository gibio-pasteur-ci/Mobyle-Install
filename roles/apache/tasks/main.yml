---
- name: Install apache
  yum: name=httpd state=installed
  notify: start httpd


- name: Ensure apache is running (and enable it at boot)
  service: name=httpd state=started enabled=yes



