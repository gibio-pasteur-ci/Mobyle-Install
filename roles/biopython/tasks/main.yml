---

- name: BioPython Install
  become: yes
  become_user: mobyle
  command: ~/bin/easy_install biopython


