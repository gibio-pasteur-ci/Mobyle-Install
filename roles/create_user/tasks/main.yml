---
- name: Creation of "mobyle" user
  user: name=mobyle comment="Application user" shell=/bin/bash state=present

- name: register mobyle_home
  become: yes
  become_user: mobyle
  shell: grep '^mobyle' /etc/passwd | cut -d ':' -f6
  changed_when: false
  register: mobyle_home

- name: Create src directory
  become: yes
  become_user: mobyle
  file: path=~/src/ state=directory

- name: Modify PATH
  become: yes
  become_user: mobyle
  lineinfile: dest=~/.bashrc insertafter=EOF line="export PATH=$HOME/bin:$PATH"