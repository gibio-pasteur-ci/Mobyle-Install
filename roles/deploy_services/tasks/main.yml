---
- name: check presence of Mobyle programs interfaces
  become: yes
  become_user: "{{ mobyle_user }}"
  stat: path=~/src/Programs-5.1.2.tgz
  register: mobyle_programs_archive

- name: download programs interfaces
  become: yes
  become_user: "{{ mobyle_user }}"
  when: mobyle_programs_archive.stat.exists == False
  command: curl ftp://ftp.pasteur.fr/pub/gensoft/projects/mobyle/Programs-5.1.2.tgz -o ~/src/Programs-5.1.2.tgz

- name: Unarchive programs
  become: yes
  become_user: "{{ mobyle_user }}"
  unarchive: src=~/src/Programs-5.1.2.tgz dest=~/Mobyle/Services/ creates=~/Mobyle/Services/Programs-5.1.2 copy=no

- name: Creating link to Programs
  become: yes
  become_user: "{{ mobyle_user }}"
  file:
    src={{ mobyle_home }}/Mobyle/Services/Programs-5.1.2
    dest={{ mobyle_home }}/Mobyle/Services/Programs
    state=link

- name: check presence of Mobyle Workflows interfaces
  become: yes
  become_user: "{{ mobyle_user }}"
  stat: path=~/src/Workflows-1.1.2.tgz
  register: mobyle_workflows_archive

- name: download workflows interfaces
  become: yes
  become_user: "{{ mobyle_user }}"
  when: mobyle_workflows_archive.stat.exists == False
  command: curl ftp://ftp.pasteur.fr/pub/gensoft/projects/mobyle/Workflows-1.1.2.tgz -o ~/src/Workflows-1.1.2.tgz

- name: Unarchive workflows
  become: yes
  become_user: "{{ mobyle_user }}"
  unarchive: src=~/src/Workflows-1.1.2.tgz dest=~/Mobyle/Services/ creates=~/Mobyle/Services/Workflows-1.1.2 copy=no

- name: Creating link to Workflows
  become: yes
  become_user: "{{ mobyle_user }}"
  file:
    src={{ mobyle_home }}/Mobyle/Services/Workflows-1.1.2
    dest={{ mobyle_home }}/Mobyle/Services/Workflows
    state=link

- name: check presence of Mobyle Tutorials interfaces
  become: yes
  become_user: "{{ mobyle_user }}"
  stat: path=~/src/Tutorials-1.5.0.tar.gz
  register: mobyle_tutorials_archive

- name: download Tutorials interfaces
  become: yes
  become_user: "{{ mobyle_user }}"
  when: mobyle_tutorials_archive.stat.exists == False
  command: curl ftp://ftp.pasteur.fr/pub/gensoft/projects/mobyle/Tutorials-1.5.0.tar.gz -o ~/src/Tutorials-1.5.0.tar.gz

- name: Unarchive Tutorials
  become: yes
  become_user: "{{ mobyle_user }}"
  unarchive: src=~/src/Tutorials-1.5.0.tar.gz dest=~/Mobyle/Services/ creates=~/Mobyle/Services/Tutorials-1.5.0 copy=no

- name: Creating link to Tutorials
  become: yes
  become_user: "{{ mobyle_user }}"
  file:
    src={{ mobyle_home }}/Mobyle/Services/Tutorials-1.5.0
    dest={{ mobyle_home }}/Mobyle/Services/Tutorials
    state=link

- name: deploying services
  become: yes
  become_user: "{{ mobyle_user }}"
  shell: mobdeploy deploy

  