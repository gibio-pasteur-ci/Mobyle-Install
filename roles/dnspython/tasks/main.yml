---

- name: dnspython Download
  become: yes
  become_user: mobyle
  get_url: url=https://pypi.python.org/packages/source/d/dnspython/dnspython-1.12.0.zip#md5=bcfab8f4f4a2280f875e646502b23e66 dest=~/src/

- name: dnspython Unarchive
  become: yes
  become_user: mobyle
  unarchive: src=~/src/dnspython-1.12.0.zip dest=~/src creates=~/src/dnspython-1.12.0 copy=no

- name: dnspython Build
  become: yes
  become_user: mobyle
  command: ~/bin/python setup.py build chdir=~/src/dnspython-1.12.0

- name: dnspython Install
  become: yes
  become_user: mobyle
  command: ~/bin/python setup.py install chdir=~/src/dnspython-1.12.0

