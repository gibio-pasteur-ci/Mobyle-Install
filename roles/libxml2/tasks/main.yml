---
- name: register mobyle_home
  become: yes
  become_user: "{{ mobyle_user }}"
  shell: grep '^mobyle' /etc/passwd | cut -d ':' -f6
  changed_when: false
  register: mobyle_home

- name: libxml2 Download
  become: yes
  become_user: "{{ mobyle_user }}"
# because of bug https://github.com/ansible/ansible-modules-core/issues/3661
# on ansible version 2.0.2.0 we can not use get_url : workaround use command
# get_url: url=ftp://xmlsoft.org/libxml2/libxml2-2.9.3.tar.gz dest=~/src/
  command: curl ftp://xmlsoft.org/libxml2/libxml2-2.9.3.tar.gz -o ~/src/libxml2-2.9.3.tar.gz

- name: libxml2 Extract
  become: yes
  become_user: "{{ mobyle_user }}"
  unarchive: src=~/src/libxml2-2.9.3.tar.gz dest=~/src creates=~/src/libxml2-2.9.3 copy=no

- name: libxml2 configure
  become: yes
  become_user: "{{ mobyle_user }}"
  command: ./configure --prefix={{mobyle_home.stdout}} --without-python chdir=~/src/libxml2-2.9.3 creates=Makefile

- name: libxml2 make
  become: yes
  become_user: "{{ mobyle_user }}"
  command: make chdir=~/src/libxml2-2.9.3

- name: libxml2 Install
  become: yes
  become_user: "{{ mobyle_user }}"
  command: make install chdir=~/src/libxml2-2.9.3
