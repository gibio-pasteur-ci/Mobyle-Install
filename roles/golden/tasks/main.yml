---
- name: check presence of golden archive
  become: yes
  become_user: "{{ mobyle_user }}"
  stat: path=~/src/golden-1.1a.tar.gz
  register: golden_archive

- name: Golden Download
  become: yes
  become_user: "{{ mobyle_user }}"
  when: golden_archive.stat.exists == False
# because of bug https://github.com/ansible/ansible-modules-core/issues/3661
# on ansible version 2.0.2.0 we can not use get_url : workaround use command
#  get_url: url=ftp://ftp.pasteur.fr/pub/gensoft/projects/golden/golden-1.1a.tar.gz dest=~/src/
  command: curl ftp://ftp.pasteur.fr/pub/gensoft/projects/golden/golden-1.1a.tar.gz -o ~/src/golden-1.1a.tar.gz

- name: Golden Unarchive
  become: yes
  become_user: "{{ mobyle_user }}"
  unarchive: src=~/src/golden-1.1a.tar.gz dest=~/src creates=~/src/golden-1.1a copy=no

- name: Golden configure
  become: yes
  become_user: "{{ mobyle_user }}"
  command: ./configure --prefix={{ mobyle_home }} chdir=~/src/golden-1.1a creates=Makefile

- name: Golden make
  become: yes
  become_user: "{{ mobyle_user }}"
  command: make chdir=~/src/golden-1.1a

- name: Golden Install
  become: yes
  become_user: "{{ mobyle_user }}"
  command: make install chdir=~/src/golden-1.1a

- name: create link to golden indexes
  become: yes
  become_user: "{{ mobyle_user }}"
  file:
    src={{ bank_root }}/index/golden
    dest=share/golden
    state=link
    force=yes
